Source/RxSwift/ObservableType+Weak.swift

```diff
@@ -29,6 +29,28 @@ extension ObservableType {
             method?(obj)()
         }
     }
+    
+    /**
+     Performs a flatMap providing a safe unretained reference to `obj` as an argument of the flatMap closure
+     
+     - parameter weak: The object that we want to have a safe and unretained reference on.
+     - parameter selector: A function that produces a new ObservableConvertibleType.
+     - returns: An observable sequence whose elements are the result of invoking the one-to-many transform function on each element of the input sequence.
+     */
+    public func flatMap<A: AnyObject, O: ObservableConvertibleType>(weak obj: A, _ selector: @escaping (A, Self.E) throws -> O) -> Observable<O.E> {
+        // swiftlint:disable force_cast
+        // Force casts are unavoidable here, because of Swift's type system limitation
+        return flatMap { [weak obj] value -> O in
+            guard let strongObj = obj else { return Observable<O.E>.empty() as! O }
+            
+            do {
+                return try selector(strongObj, value)
+            } catch {
+                return Observable<O.E>.error(error)  as! O
```

Comment:
two spaces before `as!` should be a single space.

Reply:
done